docs/xks/operator-guide/agents.md

```diff
@@ -0,0 +1,306 @@
+  "subscriptionId": "00000000-0000-0000-0000-000000000000",
+  "tenantId": "00000000-0000-0000-0000-000000000000"}
+```
+
+## VM Image
+
+We make use of [Packer](https://www.packer.io/) to create the VM images. Packer allows for the automation of the process, creating the VM snapshot for us. The VM image has to be created before any VM
+image can be created.
+
+Create a repository called `packer` that is going to contain the CI jobs that will build the VM images. Doing this will allow for tracking of versions and automate the complicated build process.
+
+There are templates for Azure DevOps that can be used to build the VM images for the agents. The following pipeline definition should be commited to the file `.ci/azure-pipelines-agent.yaml` in the
+new packer repository. After that is done create a Azure DevOps pipeline for the given pipeline definition.
+
+```yaml
+name: $(Build.BuildId)
+
+trigger: none
+
+resources:
+  repositories:
+    - repository: templates
+      type: git
+      name: XKS/azure-devops-templates
+      ref: refs/tags/2020.12.5
+
+stages:
+  - template: packer-docker/main.yaml@templates
+    parameters:
+      poolNameTemplate: ""
+      azureSubscriptionTemplate: "xks-{0}-owner"
+      resourceGroupTemplate: "rg-{0}-we-azpagent"
+      packerTemplateRepo: "https://github.com/XenitAB/packer-templates.git"
+      packerTemplateRepoBranch: "2021.06.1"
+      packerTemplateFile: "templates/azure/azure-pipelines-agent/azure-pipelines-agent.json"
+```
+
+There is also a template for GitHub that can be used for building with Packer. The following pipeline definition should be commited to the file `.github/workflows/github-runner.yaml` in the packer
+repository.
+
+```yaml
+name: packer_github_runner
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  packer:
+    uses: xenitab/azure-devops-templates/.github/workflows/packer-docker.yaml@2021.11.1
+    with:
+      ENVIRONMENTS: |
+        {
+          "environments":[
+            {"name":"prod"}
+          ]
+        }
+      RESOURCE_GROUP_NAME_SUFFIX: "ghrunner"
+      PACKER_TEMPLATE_REPO: "https://github.com/XenitAB/packer-templates.git"
+      PACKER_TEMPLATE_REPO_BRANCH: "2021.06.1"
+      PACKER_TEMPLATE_FILE: "templates/azure/azure-pipelines-agent/azure-pipelines-agent.json"
+    secrets:
+      AZURE_CREDENTIALS_PROD: ${{ secrets.AZURE_CREDENTIALS_PROD }}
+```
+
+Start the Packer build pipeline and allow it to run until completion. This may take up to 40 minutes to run so give it time. Afte the build is completed a new VM image should be created and stored the
+agents resource group in Azure.
+
+The name of the image is dynamic and includes a timestamp to allow versioning of the images. The following Azure CLI command gets the name of the image.
+
+```shell
+# Assuming that you don't have any other image this RG.
+az image list -o json --query '[0].name'
+```
+
+The name should be similar to `azp-agent-2021-04-09T08-18-30Z`.
+
+## Pre Setup
```

Comment:
Nothing says under Pre setup, remove?
Or is Guthub supposed to be under Pre Setup
If so it should have ###